services/interactions/helpers.js

```diff
@@ -142,14 +174,64 @@ export const handleConnection = async (userID, connID, timestamp) => {
       })
   };
 
-  switch (type) {
-  case ATTENDEE:
+  switch (connType) {
+  case EXEC + EXEC:
+    try {
+      await incrementQuestProgress(
+        connData.registrationID,
+        QUEST_CONNECT_EXEC_H
+      );
+    } catch (error) {
+      console.error(error);
+      return handlerHelpers.createResponse(500, {
+        message: "Internal server error"
+      });
+    }
+
+  case EXEC:
+    try {
+      await incrementQuestProgress(
+        userData.registrationID,
+        QUEST_CONNECT_EXEC_H
+      );
+    } catch (error) {
+      console.error(error);
+      return handlerHelpers.createResponse(500, {
+        message:
+            "Internal server error, data inconsistency :( not all quests progressed"
+      });
+    }
+
+    // case ATTENDEE:
+  default:
     try {
       // potential race condition -> use transactions to fix, but will take time to implement
       await db.put(connPut, CONNECTIONS_TABLE, true);
       await db.put(userPut, CONNECTIONS_TABLE, true);
-      // logic to check if quests entry has been made for connection
-      // put command to update the quest entry
+      await incrementQuestProgress(
+        userData.registrationID,
+        QUEST_CONNECT_ONE
+      );
+      await incrementQuestProgress(
+        userData.registrationID,
+        QUEST_CONNECT_FOUR
+      );
+      await incrementQuestProgress(
+        userData.registrationID,
+        QUEST_CONNECT_TEN_H
+      );
+      await incrementQuestProgress(
+        connData.registrationID,
+        QUEST_CONNECT_ONE
+      );
+      await incrementQuestProgress(
+        connData.registrationID,
+        QUEST_CONNECT_FOUR
+      );
+      await incrementQuestProgress(
+        connData.registrationID,
+        QUEST_CONNECT_TEN_H
+      );
```

Comment:
We can do all these asynchronous calls concurrently, it'll likely improve latency:

```js
try {
  const promises = [
    incrementQuestProgress(userData.registrationID, QUEST_CONNECT_ONE),
    incrementQuestProgress(userData.registrationID, QUEST_CONNECT_FOUR),
    incrementQuestProgress(userData.registrationID, QUEST_CONNECT_TEN_H),
    incrementQuestProgress(connData.registrationID, QUEST_CONNECT_ONE),
    incrementQuestProgress(connData.registrationID, QUEST_CONNECT_FOUR),
    incrementQuestProgress(connData.registrationID, QUEST_CONNECT_TEN_H)
  ];

  // Wait for all promises to resolve
  await Promise.all(promises);
} catch (error) {
  // Handle error
  console.error(error);
}
```